mobius-android/src/main/java/com/spotify/mobius/android/Accumulator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.android;
+
+import static java.util.Collections.singletonList;
+
+import com.spotify.mobius.functions.Consumer;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class Accumulator<T> {
```

Comment:
This is based on immutable (-ish, because of `handled`), but couldn't this approach make it easy to accidentally create two parallel lists, where one is missing items? I think what we really want to accumulate is a list that is guaranteed to contain all items you add to it, so maybe something less FP-style would make it easier to implement and use this class correctly?

Reply:
Yes, it is based on immutable list of events at least - and yes I can modify it to add to the current list in a more thread-safe manner too, most likely. I'll have a look to improve it.